src/domainobjects/cashflow.py

```diff
@@ -1,106 +1,207 @@
+    """ Class to generate cashflows. Generate method will generate a set
+    amount of cashflows. Other generation methods are included where cashflows
+    are the only domain object requiring them.
+
+    Process of generating cashflows dependent on Swap Positions, for each of
+    these positions, simulate whether each cashflow type (defined by the user
+    in custom arguments -> cashflow args) would accrue, and if so, generate a
+    record of such. Only Swap Positions with type 'E' (end of day) will have
+    cashflows generated.
+    """
+
+    def generate(self, record_count, start_id):
+        """ Generate a set number of cashflows
+
+        Parameters
+        ----------
+        record_count : int
+            Number of cashflows to generate
+        start_id : int
+            Starting id to generate from
+
+        Returns
+        -------
+        List
+            Containing 'record_count' cashflows
+        """
+
+        swap_position_batch =\
+            self.retrieve_batch_records('swap_positions',
+                                        record_count, start_id)
+        cashflow_gen_args = self.get_cashflow_gen_args()
+
+        records = [self.generate_record(swap_position, cf_arg)
+                   for swap_position in swap_position_batch
+                   for cf_arg in cashflow_gen_args
+                   if swap_position['position_type'] == 'E']
+        records = filter(None, records)
+        return records
+
+    def generate_record(self, swap_position, cf_arg):
+        """ Generate a single cashflow
+
+        Parameters
+        ----------
+        Swap Position : dict
+            Dictionary containing a partial record of a Swap Position, only
+            contains the information necessary to generate Cashflows
+        cf_arg : dict
+            Dictionary defining the particular cashflow being generated for
+
+        Returns
+        -------
+        dict
+            A single cashflow object
+        """
+
+        accrual = cf_arg['cashFlowAccrual']
+        probability = cf_arg['cashFlowAccrualProbability']
+        effective_date = self.effective_date(swap_position['effective_date'])
+        if self.cashflow_accrues(effective_date, accrual, probability):
+            record = self.instantiate_record()
+
+            pay_date_period = cf_arg['cashFlowPaydatePeriod']
+            p_date_func = self.get_pay_date_func(pay_date_period)
+            record['effective_date'] = effective_date
+            record['swap_contract_id'] = swap_position['swap_contract_id']
+            record['ric'] = swap_position['ric']
+            record['long_short'] = swap_position['long_short']
+            record['cashflow_type'] = cf_arg['cashFlowType']
+            record['pay_date'] = datetime.strftime(p_date_func(effective_date),
+                                                   '%Y-%m-%d')
+            record['currency'] = self.generate_currency()
+            record['amount'] = self.generate_random_integer()
```

Comment:
get a real account. May be worth splitting accounts in to (CASH, STOCK LOAN, FULLY PAID, HYPO, FINANCED) and only selecting CASH accounts for this. But may be unnecessarily complicated